src/printer.rs

```diff
@@ -315,17 +310,102 @@ impl<'a> Printer for InteractivePrinter<'a> {
         };
 
         let description = &input.description;
+        let metadata = &input.metadata;
+
+        // We use this iterator to have a deterministic order for
```

Comment:
If we go with option 1 (which I prefer we don't need a Vec, since the filename shall always go before the filesize if present.

I'm actually not sure we need to allow a custom order even if we go with option 2. But I would prefer if we went with option 1 so that we can postpone this problem to a follow-up PR.